silkrpc/commands/eth_api.cpp

```diff
@@ -157,7 +157,13 @@ boost::asio::awaitable<void> EthereumRpcApi::handle_eth_gas_price(const nlohmann
         };
 
         GasPriceOracle gas_price_oracle{ block_provider};
-        const auto gas_price = co_await gas_price_oracle.suggested_price(block_number);
+        auto gas_price = co_await gas_price_oracle.suggested_price(block_number);
+
+        const auto block_with_hash = co_await block_provider(block_number);
+        const auto &base_fee = block_with_hash.block.header.base_fee_per_gas.value_or(0);
+        if (base_fee != 0) {
+           gas_price += base_fee;
+        }
```

Comment:
Here I would suggest one of the following approaches to improve readability:

1)
```
const auto base_fee = block_with_hash.block.header.base_fee_per_gas;
if (base_fee) {
    gas_price += base_fee;
}
```

2)
```
const auto base_fee = block_with_hash.block.header.base_fee_per_gas.value_or(0);
gas_price += base_fee;
```

Reply:
I have implemented option1